src/memray/_memray/records.h

```diff
@@ -93,6 +93,7 @@ struct MemoryRecord
 {
     unsigned long int ms_since_epoch;
     size_t rss;
+    size_t heap;
```

Comment:
Hm, this is quite ugly. This same `MemoryRecord` struct is used on the tracking side and on the reader side. On the tracking side, the `heap` field is never set (or even initialized!), and it is ignored by the `RecordWriter`. On the reader side, the `RecordReader` returns a `MemoryRecord` whose `heap` field is uninitialized to the Cython layer, and the Cython layer then copies the two valid fields out of that `MemoryRecord` into a new `MemoryRecord`, ignoring the uninitialized and invalid `heap` value, and provides a valid value for `heap` that it computed independently.

I think we really want two different structures instead - I think `MemoryRecord` should not get a `heap` field added to it, and the `vector[_MemoryRecord]` in the Cython layer should become a vector of a new type that has the additional field.

Reply:
Whops, sorry, this was actually done in another commit but I messed it up :(

I have pushed a fixed version that uses a `MemorySnapshot` to not interfere with the native side.